examples/scripts/run_tests.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+import sys
+import pathlib
+import argparse
+import subprocess
+
+HERE = pathlib.Path(__file__).parent
+EXAMPLES_ROOT = HERE.parent
+CASEMATE_ROOT = EXAMPLES_ROOT.parent
+CASEMATE_CHECK_ROCQ_ROOT = CASEMATE_ROOT / "src" / "casemate-check-rocq"
+
+EXAMPLES = (
+    subprocess.run(
+        ["make", "list-build-objs"],
+        cwd=EXAMPLES_ROOT,
+        capture_output=True,
+        text=True,
+        check=True,
+    ).stdout.strip().split()
+)
+
+
+def runmsg(prefix, s):
+    print(f'  {prefix:<8}\t\t\t{s}', file=sys.stderr)
+
+def check_expected(test_name):
+    example_exe = EXAMPLES_ROOT / test_name
+    out_path = (EXAMPLES_ROOT / "tests" / test_name).with_suffix(".log")
+
+    runmsg("RUN", test_name)
+    with open(out_path, "wb") as logf:
+        subprocess.run(
+            [str(example_exe)],
+            cwd=EXAMPLES_ROOT,
+            stdout=logf,
+            check=False,
+        )
+
+    expected = (EXAMPLES_ROOT / "expected" / test_name).with_suffix(".log")
+    runmsg("CHECK", test_name)
+    subprocess.run(
+        ["python3", "./scripts/check_simulation.py", str(out_path), str(expected)],
+        cwd=EXAMPLES_ROOT,
+        check=True,
+    )
+
+def check_rocq_trace(test_name):
+    expected_log = (EXAMPLES_ROOT / "expected" / test_name).with_suffix(".log")
+
+    runmsg("CHECK", test_name)
+
+    expected_status = expected_log.read_text().strip().splitlines()[-1]
+    expected_status = 121 if expected_status.startswith("!") else 0
```

Comment:
This looks good for now, but maybe we can enhance the code later to check whether the error messages from the failing tests align with the expected errors.